src/internal/utils/node-belongs.ts

```diff
@@ -10,11 +10,11 @@ import { containsOrEqual, findUpUntil } from './dom';
  * @param container Container node
  * @param target Node that is checked to be a descendant of the container
  */
-export function nodeBelongs(container: Node | null, target: Node): boolean {
-  const portal = findUpUntil(
-    target as HTMLElement,
-    node => node instanceof HTMLElement && !!node.dataset.awsuiReferrerId
-  );
+export function nodeBelongs(container: Node | null, target: Node | EventTarget | null): boolean {
+  if (!(target instanceof Node)) {
+    return false;
+  }
+  const portal = findUpUntil(target as HTMLElement, node => !!node.dataset.awsuiReferrerId);
```

Comment:
This type assertion (*`as HTMLElement`) is not safe either, right? We can't guarantee that `nodeBelongs` will never be called on e.g. an SVG element. Can we replace the type assertion with a check?

Reply:
Tried this while working on this change: https://github.com/cloudscape-design/components/compare/3d036443c0771b25e3143449f7cdbf65156289c1..550b17351b54911a505868360a7d05a5e4dc6bdb

It broke some logic in area chart, where we use this utility with SVG nodes. 


Fixing this properly would require refactoring `findUpUntil` which stretches the scope of this work a bit. 

Helping me fixing this part of our codebase would be also appreciated.